### PR TITLE
CSP-1389 Get Permissions from ukprn and hashedId

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.PR.Application.UnitTests/AccountProviderLegalEntities/Queries/GetAccountProviderLegalEntities/GetAccountProviderLegalEntitiesQueryValidatorTests.cs [hunk 1]
```diff
@@ -47,6 +47,6 @@ public async Task GetAccountProviderLegalEntitiesQueryValidator_Validate_Operati
         var sut = new GetAccountProviderLegalEntitiesQueryValidator();
         var result = await sut.TestValidateAsync(query);
         result.ShouldHaveValidationErrorFor(q => q.Operations)
-                    .WithErrorMessage(GetAccountProviderLegalEntitiesQueryValidator.OperationFilterValidationMessage);
+                    .WithErrorMessage(OperationsValidator.OperationFilterValidationMessage);
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.PR.Application.UnitTests/AccountProviderLegalEntities/Queries/GetAccountProviderLegalEntities/GetAccountProviderLegalEntitiesQueryValidatorTests.cs:50]: Yes. It's broken in main, this fixes it

File: src/SFA.DAS.PR.Data/Repositories/PermissionsReadRepository.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    private readonly IProviderRelationshipsDataContext _providerRelationshipsDataContext = providerRelationshipsDataContext;
+
+    public async Task<List<Operation>> GetOperations(long ukprn, string publicHashedId, CancellationToken cancellationToken)
+    {
+        var permissions = await _providerRelationshipsDataContext.Permissions
+            .AsNoTracking()
+            .Include(p => p.AccountProviderLegalEntity)
+            .ThenInclude(a => a.AccountProvider)
+            .Include(x => x.AccountProviderLegalEntity)
+            .ThenInclude(y => y.AccountLegalEntity)
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.PR.Data/Repositories/PermissionsReadRepository.cs:17]: ah didn't know that. I assumed as the tables are needed in the query, they would need to be included. Will remove.

File: src/SFA.DAS.PR.Data/Repositories/PermissionsReadRepository.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            .Where(x => x.AccountProviderLegalEntity.AccountProvider.ProviderUkprn == ukprn
+                        && x.AccountProviderLegalEntity.AccountLegalEntity.PublicHashedId == publicHashedId)
+            .ToListAsync(cancellationToken);
+
+        return permissions.Select(x => x.Operation).ToList();
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.PR.Data/Repositories/PermissionsReadRepository.cs:22]: ok

File: src/SFA.DAS.PR.Api/Controllers/PermissionsController.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+[Route("permissions")]
+public class PermissionsController(IMediator _mediator) : ActionResponseControllerBase
+{
+    public override string ControllerName => "Permissions";
+
+    [HttpGet()]
+    [Authorize]
+    [ProducesResponseType(typeof(bool), StatusCodes.Status400BadRequest)]
+    [SwaggerRequestExample(typeof(PermissionsRouteValues), typeof(GetPermissionsQueryResultExample))]
+    public async Task<IActionResult> GetPermissions([FromQuery] PermissionsRouteValues routeValues, CancellationToken cancellationToken)
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.PR.Api/Controllers/PermissionsController.cs:22]: changed to use the query

File: src/SFA.DAS.PR.Api/SwaggerExamples/GetPermissionsQueryResultExample.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
+﻿using SFA.DAS.PR.Application.AccountProviders.Queries.GetPermissions;
+using Swashbuckle.AspNetCore.Filters;
+using System.Diagnostics.CodeAnalysis;
+
+namespace SFA.DAS.PR.Api.SwaggerExamples;
+
+[ExcludeFromCodeCoverage]
+public class GetPermissionsQueryResultExample : IExamplesProvider<PermissionsRouteValues>
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.PR.Api/SwaggerExamples/GetPermissionsQueryResultExample.cs:8]: yep, I don't know why I made the request the focus.  Changed to be result

File: src/SFA.DAS.PR.Api/Controllers/PermissionsController.cs [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+[ApiController]
+[Route("permissions")]
+public class PermissionsController(IMediator _mediator) : ActionResponseControllerBase
+{
+    public override string ControllerName => "Permissions";
+
+    [HttpGet()]
+    [Authorize]
+    [ProducesResponseType(typeof(bool), StatusCodes.Status400BadRequest)]
+    [SwaggerRequestExample(typeof(PermissionsRouteValues), typeof(GetPermissionsQueryResultExample))]
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.PR.Api/Controllers/PermissionsController.cs:21]: I've put one in

File: src/SFA.DAS.PR.Api.UnitTests/Controllers/PermissionsControllerGetTests.cs [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        mediatorMock.Verify(m =>
+            m.Send(It.Is<GetPermissionsQuery>(q =>
+                    q.Ukprn == routeValues.Ukprn
+                    && q.PublicHashedId == routeValues.PublicHashedId),
+            cancellationToken)
+        );
+    }
+
+    [Test, MoqAutoData]
+    public async Task HasPermission_HandlerReturnsNullResult_ReturnsNotFoundResponse(
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.PR.Api.UnitTests/Controllers/PermissionsControllerGetTests.cs:35]: ok, removed

File: src/SFA.DAS.PR.Api/Controllers/PermissionsController.cs [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace SFA.DAS.PR.Api.Controllers;
+
+[ApiController]
+[Route("permissions")]
+public class PermissionsController(IMediator _mediator) : ActionResponseControllerBase
+{
+    public override string ControllerName => "Permissions";
+
+    [HttpGet()]
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.PR.Api/Controllers/PermissionsController.cs:19]: ok

File: src/SFA.DAS.PR.Data/Repositories/PermissionsReadRepository.cs [hunk 9]
```diff
@@ -0,0 +1,24 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SFA.DAS.PR.Domain.Entities;
+using SFA.DAS.PR.Domain.Interfaces;
+
+namespace SFA.DAS.PR.Data.Repositories;
+public class PermissionsReadRepository(IProviderRelationshipsDataContext providerRelationshipsDataContext) : IPermissionsReadRepository
+{
+    private readonly IProviderRelationshipsDataContext _providerRelationshipsDataContext = providerRelationshipsDataContext;
+
+    public async Task<List<Operation>> GetOperations(long ukprn, string publicHashedId, CancellationToken cancellationToken)
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.PR.Data/Repositories/PermissionsReadRepository.cs:10]: ok added

File: src/SFA.DAS.PR.Api/Controllers/PermissionsController.cs [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    [Authorize]
+    [ProducesResponseType(typeof(bool), StatusCodes.Status400BadRequest)]
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.PR.Api/Controllers/PermissionsController.cs]: fixed

File: src/SFA.DAS.PR.Data.UnitTests/Permissions/Queries/PermissionsReadRepositoryTests.cs [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+public class PermissionsReadRepositoryTests
+{
+    private readonly CancellationToken _cancellationToken = CancellationToken.None;
+
+    private Fixture _fixture = null!;
+
+    [SetUp]
+    public void Setup()
+    {
+        _fixture = new Fixture();
```
Comment (DevenKShah) [inline src/SFA.DAS.PR.Data.UnitTests/Permissions/Queries/PermissionsReadRepositoryTests.cs:18]: _fixture is not used anywhere

**Review by DevenKShah — APPROVED** [review]

**Review by DevenKShah — APPROVED** [review]

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**